ocaml/xapi/xapi_xenops.ml

```diff
@@ -2388,14 +2389,14 @@ let events_from_xapi () =
                 (* We register for events on resident_VMs only *)
                 let resident_VMs = Db.Host.get_resident_VMs ~__context ~self:localhost in
 
-                let uuids = List.map (fun self -> Db.VM.get_uuid ~__context ~self) resident_VMs in
-                let cached = Mutex.execute metadata_m (fun () -> Xenops_cache.list_nolock ()) in
-                let missing_in_cache = Listext.List.set_difference uuids cached in
-                let extra_in_cache = Listext.List.set_difference cached uuids in
-                if missing_in_cache <> []
-                then error "events_from_xapi: missing from the cache: [ %s ]" (String.concat "; " missing_in_cache);
-                if extra_in_cache <> []
-                then error "events_from_xapi: extra items in the cache: [ %s ]" (String.concat "; " extra_in_cache);
+                let uuids = StringSet.of_list (List.rev_map (fun self -> Db.VM.get_uuid ~__context ~self) resident_VMs) in
+                let cached = StringSet.of_list (Mutex.execute metadata_m (fun () -> Xenops_cache.list_nolock ())) in
+                let missing_in_cache = StringSet.diff uuids cached in
+                let extra_in_cache = StringSet.diff cached uuids in
+                if missing_in_cache <> StringSet.empty
```

Comment:
There is a StringSet.is_empty you could use here instead, usually it is best to avoid polymorphic comparisons on sets (the empty set would probably work, but comparing otherwise identical sets with polymorphic compare could result in them being considered different).

Reply:
Ok, I've changed this to use the `is_empty` function instead

Reply:
Sorry I didn't mean to literally use `String.is_empty`, obviously preserve the original meaning of the test and check that it is not empty by negating it.

Reply:
Yes, it was a mistake, Christian also pointed it out